Core/Src/monitor.c

```diff
@@ -51,7 +58,26 @@ void read_lv_sense(void *arg)
 	// get final voltage
 	v_int = (uint32_t)(v_dec * 10.0);
 
-	memcpy(msg.data, &v_int, msg.len);
+	// Calculate SoC using logistic function
+	// - Normal charged voltage is 29.4V
+	// - 18650 max charged voltage is 4.2V
+	// - 29.4V/4.2V = 7 batteries
+	// - Divide v_dec by 7 to get avg voltage over all rows
+	// - Use logistic function to model SoC
+	// https://www.aegisbattery.com/products/24v-20ah-li-ion-battery-pvc
+	float v_max = 4.2; // max avg voltage over all rows (7)
+	float v_min = 2.8; // min avg voltage over all rows (7)
+	float k = -8.5; // logistic fn parameter to affect steepness of curve
+	float i = (v_max + v_min) /
+		  2; // logistic fn parameter to affect midpoint of curve
+	float soc_dec =
+		1 / (1 + exp(k * (v_dec - i))); // SoC calculation in [0,1]
+	soc_int = soc_dec * 100;
```

Comment:
so we apply a 100 scale to this 0 to 1 value, so the data sent over CAN is casted 0 to 100.  We can do to things to clarify this:
- Add a prescaler and corresponding JSON postscaler which allows us to get decimal points out of our SoC
- Or: Use a uint8_t instead as this is a number 0 to 100 so mine as well save space.